conf/cassandra.yaml

```diff
@@ -661,6 +661,23 @@ auto_snapshot: true
 tombstone_warn_threshold: 1000
 tombstone_failure_threshold: 100000
 
+# Filtering and secondary index queries at read consistency levels above ONE/LOCAL_ONE use a
+# mechanism called replica filtering protection to ensure that results from stale replicas do
+# not violate consistency. This mechanism, which operates in two phases, caches replica
+# results on-heap at the coordinator. The more stale results returned by the former, the more
+# rows cached at the latter.
+#
+# These thresholds exist to limit the damage very stale replicas can cause during these queries.
+# They essentially define the memory budget individal index and filtering queries have to meet
+# the client-desired consistency level.
+#
+# "cached_replica_rows_warn_threshold" is the threshold at which a warning will be logged.
+# "cached_replica_rows_fail_threshold" is the threshold at which the query will fail.
+#
+# These thresholds may also be adjusted at runtime using the StorageService mbean.
+cached_replica_rows_warn_threshold: 1024
+cached_replica_rows_fail_threshold: 16384
```

Comment:
As always, I'm open to suggestions around naming and how I've described this.

Reply:
Definitely agree on throwing the ticket number in there for reference.

RE "cached" vs. "materialized", is the danger that readers will perceive the former as meaning rows are cached past the lifetime of the individual query? I'm definitely okay w/ changing it to "materialized" in the description, although the name is starting to get really long. What would you say to something like `materialized_rows_warn_threshold`. It isn't the most specific name, but it's also not less specific than `tombstone_warn_threshold` :/

Reply:
Given the existing usage of materialised on the project it might be best to pick something else again :/

Reply:
>RE "cached" vs. "materialized", is the danger that readers will perceive the former as meaning rows are cached past the lifetime of the individual query?

Yes, right, although I might be wrong. `cached_filtered_rows_warn_threshold` or `materialized_filtered_rows_warn_threshold` also feels a bit more descriptive to me, underlying the relationship with filtering. The rows are cached/materialized in the coordinator, and perhaps one might think the opposite with that "replica_rows".

>Given the existing usage of materialised on the project it might be best to pick something else again :/

Is that because readers might confuse materialized replica/filtered rows with materialized views? That could happen using the same adjective, I guess. Perhaps "loaded_filtered_rows", or "accumulated"? I have no better ideas, so it's ok for me if we go with "cached".

Reply:
I like `cached_filtered_rows_warn_threshold`. It at least avoids confusion about the rows being cached at the replica instead of the coordinator. If we want to make that explicit, `coordinator_cached_rows_warn_threshold` could work, but I'd rather have this hint at the relationship w/ filtering. I'll go with `cached_filtered_rows_warn_threshold` if there are no other objections.

Reply:
`temporary` rather than `cached` perhaps?

I've not looked closely at the semantics, honestly, so just throwing it out there.  I often look in a thesaurus for inspiration, otherwise, when struggling!

Reply:
At minimum, we'll need to explain that this is scoped to a (single page of) a query.

I'll throw one more crazy idea (probably overkill) out there before I proceed with `cached_filtered_rows_warn_threshold` and a clear description of its per-query-ness: a namespace.

Example: 

```
replica_filtering_protection:
    cached_rows_warn_threshold: 1000
    cached_rows_fail_threshold: 16000
```

Reply:
honestly we should do this with more parameters - quick, muck things up on 15234 with this suggestion (actually I'm serious - this might clean up our config file a lot)

Reply:
@adelapena Unless you object, I think I'll try the namespace format from my last comment. Once that's in place, I think this is ready for a more official review.

Reply:
I really like the nested properties approach. Adding the `replica_filtering_protection` section gives context to the properties and avoids any confusion with row caches, materialized views, etc., while also keeping the name length short.